File: gaia-sdk-javascript/src/graphql/response/type/UpdatedSkillImpulse.ts

import {Skill} from "./Skill";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update Skill impulse
*/
export interface UpdatedSkillImpulse {
    id?:Uuid, 
    /**
    * the Skill instance
    */
    data?:Skill
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/UpdatedSkillImpulse.py

from gaia_sdk.graphql.request.type.Skill import Skill

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField


class UpdatedSkillImpulse(list):
    """
    Impulse which indicates the result of a update Skill impulse
    """

    def id(self):
        self.append(lambda x: "id")

    """
    the Skill instance
    """
    def data(self, config: Callable[['Skill'], None]):
        def callback(registry: VariableRegistry):
            entity = Skill()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))